pub_sub/javascript/sdk/infra/resources.bicep

```diff
@@ -6,179 +6,93 @@ param tags object
-        customerId: logAnalyticsWorkspace.properties.customerId
-        sharedKey: logAnalyticsWorkspace.listKeys().primarySharedKey
-      }
-    }
-  }
-
-  resource daprComponent 'daprComponents@2022-03-01' = {
-    name: 'orderpubsub'
-    properties: {
-      componentType: 'pubsub.azure.servicebus'
-      version: 'v1'
-      secrets: [
-        {
-          name: 'sb-root-connectionstring'
-          value: '${listKeys('${sb.id}/AuthorizationRules/RootManageSharedAccessKey', sb.apiVersion).primaryConnectionString};EntityPath=orders'
-        }
-      ]
-      metadata: [
-        {
-          name: 'connectionString'
-          secretRef: 'sb-root-connectionstring'
-        }
-        {
-          name: 'consumerID'
-          value: 'orders'
-        }
-      ]
-      scopes: []
-    }
-  }
-
-}
-
-resource containerRegistry 'Microsoft.ContainerRegistry/registries@2021-12-01-preview' = {
-  name: 'contreg${resourceToken}'
-  location: location
-  tags: tags
-  sku: {
-    name: 'Standard'
-  }
-  properties: {
-    adminUserEnabled: true
-    anonymousPullEnabled: false
-    dataEndpointEnabled: false
-    encryption: {
-      status: 'disabled'
-    }
-    networkRuleBypassOptions: 'AzureServices'
-    publicNetworkAccess: 'Enabled'
-    zoneRedundancy: 'Disabled'
-  }
-}
-
-resource keyVault 'Microsoft.KeyVault/vaults@2019-09-01' = {
-  name: 'keyvault${resourceToken}'
-  location: location
-  tags: tags
-  properties: {
-    tenantId: subscription().tenantId
-    sku: {
-      family: 'A'
-      name: 'standard'
-    }
-    accessPolicies: []
-  }
-
 }
 
-resource keyVaultAccessPolicies 'Microsoft.KeyVault/vaults/accessPolicies@2021-11-01-preview' = if (!empty(principalId)) {
-  name: '${keyVault.name}/add'
-  properties: {
-    accessPolicies: [
-      {
-        objectId: principalId
-        permissions: {
-          secrets: [
-            'get'
-            'list'
-          ]
-        }
-        tenantId: subscription().tenantId
-      }
-    ]
+module keyVaultResources './keyvault.bicep' = {
+  name: 'keyvault-resources'
+  params: {
+    location: location
+    tags: tags
+    resourceToken: resourceToken
+    principalId: principalId
   }
 }
 
-resource sb 'Microsoft.ServiceBus/namespaces@2018-01-01-preview' existing = {
-  name: serviceBusResources.name
-}
-
 module serviceBusResources './servicebus.bicep' = {
-  name: 'sb-${resourceToken}'
+  name: 'sb-resources'
   params: {
-    resourceToken: resourceToken
     location: location
+    tags: tags
+    resourceToken: resourceToken
     skuName: 'Standard'
   }
 }
 
 module appInsightsResources './appinsights.bicep' = {
-  name: 'appinsights-${resourceToken}'
+  name: 'appinsights-resources'
   params: {
-    resourceToken: resourceToken
     location: location
     tags: tags
+    resourceToken: resourceToken
   }
 }
 
-module checkout './checkout.bicep' = {
-  name: 'api-resources-${resourceToken}'
+module checkoutResources './checkout.bicep' = {
```

Comment:
I'm not sure why we have this, if we now have the containerapps.bicep / containerappsresources..  I'd love to understand this.

Reply:
each containerapp needs to live in its own bicep file so it can be individually provisioned and deployed without having to reprovision the entire solution.